setup.py

```diff
@@ -124,7 +124,7 @@
     ],
     entry_points={"console_scripts": ["allennlp=allennlp.run:run"]},
     setup_requires=setup_requirements,
-    tests_require=["pytest", "flaky", "responses>=0.7"],
+    tests_require=["pytest", "flaky", "responses>=0.7", "semantic_version"],
```

Comment:
Think you either need to add this to dev-requirements, or above. I'm not actually sure why we maintain tests_require given that one of our allennlp commands run the tests...

Reply:
Thanks for catching this, fixed. Yeah, I don't know enough about `setuptools` to say. Added a TODO for when someone has the time to grok that.

Reply:
Yeah, actually `tests_require` is only necessary when using `python setup.py test`, which is actually deprecated. So this entry can be removed.

Reply:
Thanks I modified the comment to mention that, but left the line in case someone is still depending on it.

Reply:
I don't think something depends on it because I have never seen a reference to `python setup.py test` in this codebase. But I think it's okay if you leave it, yeah.

Reply:
Because AllenNLP is a library determining whether someone else has a dependency on some feature is not tractable. For this reason I try to avoid unnecessary breaking changes.

Reply:
I see, makes sense. However, `python setup.py test` doesn't apply to people that do `pip install` because there are no tests in the package unless they do `--editable`, or clone/copy the codebase and explicitly run the tests this way, which has been deprecated by setuptools a while ago.

Reply:
Also, we're preparing for 1.0.0, so I'd take advantage and break compatibility now rather than after its release.

Reply:
I'm not sure what deleting that line gains. But it could well break a script somebody out there wrote. I'm fine with breaking backward compatibility when there is a gain, but I don't see that here. Just because we're preparing for 1.0.0 doesn't mean we should break backward compatibility without a clear benefit.

Reply:
Just that we have a duplicate definition of test reqs, and this one is actually inaccurate, cause also the rest of the things in `dev-requirements.txt` should be necessary as well.